### PR TITLE
Function to_utf8 supports char type

### PR DESCRIPTION
**Pull request by borderlayout**
## Description
Function to_utf8 doesn't support char type. When this function is applied to the char type in Hive, an error will be reported. This PR modifies to_utf8 function to support  char type in Hive. This function is important, because it also used in ranger-trino plugin for masking.

## Additional context and related issues
In hive, create a table using SQL: 
CREATE TABLE char_table_test (
    id INT,
    name string,
    description CHAR(120)
);
insert into char_table_test values(1, 'xiaoming', 'he is a student');
insert into char_table_test values(2, 'honghong', 'she is a teacher');  

When using function to_utf8 in a sql, there is an error. 
To_utf8 doesn't support char type, as below:
**SQL1:**
select to_utf8(description) from char_table_test;
Query 20231218_125526_00001_ewnva failed: line 1:8: Unexpected parameters (char(120)) for function to_utf8. Expected: to_utf8(varchar(x))
select to_utf8(description) from char_table_test;

**SQL2:** 
select length(from_utf8(to_utf8(description))) from char_table_test;
Query 20231218_125334_00000_ewnva failed: line 1:25: Unexpected parameters (char(120)) for function to_utf8. Expected: to_utf8(varchar(x))
select length(from_utf8(to_utf8(description))) from char_table_test

After changed, it supports char type. As below:

SQL1:
select to_utf8(description) from char_table_test;
_col0                      
-------------------------------------------------
 68 65 20 69 73 20 61 20 73 74 75 64 65 6e 74 20+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20                         
 73 68 65 20 69 73 20 61 20 74 65 61 63 68 65 72+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20+
 20 20 20 20 20 20 20 20                         
(2 rows)

SQL2:
select from_utf8(to_utf8(description)) from char_table_test;
_col0                                                           
--------------------------------------------------------------------------------------------------------------------------
 she is a teacher                                                                                                         
 he is a student         

SQL3:
select length(from_utf8(to_utf8(description))) from char_table_test;
 _col0 
-------
   120 
   120 
(2 rows)

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by mosabua** [comment]
Docs need update as well

**Comment by martint** [comment]
Syntax and semantics look good.

**Comment by borderlayout** [comment]
I have fixed a warning message(as below) by a new commit. Some checks are still not successful, it seems an environment problem.
old message:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:testCompile (default-testCompile) on project trino-main: Compilation failure: Compilation failure: 
Error:  /home/runner/work/trino/trino/core/trino-main/src/test/java/io/trino/operator/scalar/TestStringFunctions.java:[2245,39] [UnicodeEscape] Using unicode escape sequences for printable ASCII characters is obfuscated, and potentially dangerous.

File: core/trino-main/src/main/java/io/trino/operator/scalar/StringFunctions.java [hunk 1]
```diff
@@ -880,6 +881,15 @@ public static Slice toUtf8(@SqlType("varchar(x)") Slice slice)
         return slice;
     }
 
+    @Description("Encodes the string to UTF-8")
+    @ScalarFunction
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.VARBINARY)
+    public static Slice toUtf8(@LiteralParameter("x") long x, @SqlType("char(x)") Slice slice)
+    {
+        return Chars.padSpaces(slice, (int) x);
```
Comment (wendigo) [inline core/trino-main/src/main/java/io/trino/operator/scalar/StringFunctions.java:890]: toIntExact(x)

Comment (borderlayout) [reply core/trino-main/src/main/java/io/trino/operator/scalar/StringFunctions.java]: OK, I have just updated the code.

**Comment by wendigo** [comment]
@mosabua please squash these commits into a single one

**Comment by mosabua** [comment]
> @mosabua please squash these commits into a single one

I think @wendigo meant for @borderlayout to squash commits...

**Pull request merged**